### PR TITLE
[Term Entry] Go String Function: Replace()

### PR DESCRIPTION
**Pull request by frhadjmli**
### Description

- add `strings.Replace()`  term for Go
- closes issue #2632 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@frhadjmli before review I wanted to let you know there was an error in the issue path. The file should be located in `docs/content/go/concepts/strings/terms/replace/replace.md`. Sorry for the confusion.

**Comment by frhadjmli** [comment]
@SSwiniarski I correct the file location, and it's ready for your review.

**Comment by SSwiniarski** [comment]
@frhadjmli, too much of this article is copied from [this site](https://www.educative.io/answers/how-to-use-stringsreplace-in-golang). 

We take plagiarism seriously for Codecademy Docs and for Codecademy content in general because a single case of plagiarism could jeopardize the credibility of all other 650+ entries as well as our greater catalog.
Here's the Codecademy Docs [Code of Conduct](https://github.com/yangc95/docs/blob/main/.github/CODE_OF_CONDUCT.md) on GitHub.
Since this is your first occurrence, I am issuing a warning.
We truly appreciate your help with your contribution so far and we don't want to lose you temporarily or permanently. But as we attempt to build the best coding documentation on the web, it's a long-term game and we want to focus on quality, not quantity!
We hope you understand why we are taking this very seriously.

I note that this was partially rewritten, but in order for this to be reviewed, you need to rewrite _all_ the parts of this entry that are copied from the other site.

**Comment by frhadjmli** [comment]
@SSwiniarski  I made a series of changes in the file, please review it and if you have any points or find any issues, mark them in the "files changed" tab so that I can work more precisely.

**Comment by SSwiniarski** [comment]
@frhadjmli so far so good. You still need to edit the description in the metadata.

**Comment by frhadjmli** [comment]
@SSwiniarski I edit the description in the metadata.

File: content/go/concepts/strings/terms/replace/replace.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+Title: 'Replace()' 
+Description: 'replaces all or number of occurrences of a specified substring within a given string with another substring. It returns a new string with the replacements made.'
```
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/replace/replace.md:3]:
```suggestion
Description: 'Replaces all or a number of occurrences of a specified substring within a given string with another substring.' 
```

File: content/go/concepts/strings/terms/replace/replace.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - 'Strings'
+  - 'Characters'
+  - 'Functions'
+  - 'Methods'
+CatalogContent: 
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **strings.Replace()** replaces all or number of occurrences of a specified substring within a given string with another substring. It returns a new string with the replacements made.
```
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/replace/replace.md:17]:
```suggestion
The **`strings.Replace()`** replaces all or a number of occurrences of a specified substring within a given string with another substring. It returns a new string with the replacements made.
```

File: content/go/concepts/strings/terms/replace/replace.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Syntax
+
+```pseudo
+func Replace(provided str, old str, new str, n int) string
+```
+
+If the 'old str' is empty, it corresponds to the beginning of the 'provided str'. It returns up to k+1 replacements after each UTF-8 sequence, resulting in a k-rune string. If n is a negative integer, there is no limit to the number of substitutions that can be made.
```
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/replace/replace.md:25]:
```suggestion
If the `old` string is empty, it corresponds to the beginning of the `provided` string. It returns up to k+1 replacements after each UTF-8 sequence, resulting in a k-rune string. If `n` is a negative integer, there is no limit to the number of substitutions that can be made.
```

File: content/go/concepts/strings/terms/replace/replace.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+The `strings.Replace()` function takes four parameters:
+
+- `provided str`: given or the original string
+- `old str`: the string that will be replaced
+- `new str`: the string that replaces the old string
+- `n`: the count of replacements made for the old string
```
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/replace/replace.md:32]:
```suggestion
- `provided`: The original string.
- `old`: The string that will be replaced.
- `new`: The string that replaces the `old` string.
- `n`: The count of replacements made of the `old` string.
```

File: content/go/concepts/strings/terms/replace/replace.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+```
+
+The output will be:
+
+```shell
+</> cademy CODEcademy </> cademy
+!e!mpty old string
+```
+
+## Codebyte Example (if applicable)
```
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/replace/replace.md:59]:
```suggestion
## Codebyte Example 
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@frhadjmli I did a first review and left some comments.

**Comment by frhadjmli** [comment]
@SSwiniarski I made the requested changes.

**Review by SSwiniarski — APPROVED** [review]
@frhadjmli Looks good for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@frhadjmli @SSwiniarski I just made a few small edits. Looks good to go

**Comment by SSwiniarski** [comment]
@frhadjmli Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/go/strings/replace

image

You're currently anonymous. If you don't wish to be anonymous, check out this forum thread on how to link your GitHub and Codecademy profiles: https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036  **Note:** it may take a few minutes for the change to register on the entry page.

**Pull request merged**